### ISSUE TITLE
WASM instructions do not work

### ISSUE THREAD

**Issue by bradrn** [haskell/ghcup-hs#951]
Following the [instructions to install a WASM bindist](https://www.haskell.org/ghcup/guide/#ghc-wasm-cross-bindists-experimental), I got the following error:

```
$ ghcup install ghc --set wasm32-wasi-ghc-9.9.20230623 -- --host=x86_64-linux --with-intree-gmp --with-system-libffi
[ Warn  ] New GHCup version available: 0.1.20.0. To upgrade, run 'ghcup upgrade'
[ Warn  ] New ghc version available. If you want to install this latest version, run 'ghcup install ghc 9.8.1'
[ Warn  ] New cabal version available. If you want to install this latest version, run 'ghcup install cabal 3.10.2.0'
[ Warn  ] New hls version available. If you want to install this latest version, run 'ghcup install hls 2.5.0.0'
[ Error ] [GHCup-07140] Both installation and setting the tool failed. Install error was: Unable to find a download for the requested version/distro. 
[ ...   ] Set error was: The version wasm32-wasi-ghc-9.9.20230623 of the tool ghc is not installed.
[ Error ] Also check the logs in /home/bradrn/.ghcup/logs
```

Except that I *do* have `wasm32-wasi-ghc-9.9.20230623` installed:

```
$ wasm32-wasi-ghc-9.9.20230623 --version
The Glorious Glasgow Haskell Compilation System, version 9.9.20230623
```

What is causing this?

**Comment by hasufell** [comment]
You have a stray `ghc` in your version identifier.

It's `wasm32-wasi-9.6.3.20230927`.

**Comment by bradrn** [comment]
Alas, that doesn’t work either:

```
$ ghcup install ghc --set wasm32-wasi-9.9.20230623 -- --host=x86_64-linux --with-intree-gmp --with-system-libffi
[ Warn  ] New ghc version available. If you want to install this latest version, run 'ghcup install ghc 9.8.1'
[ Warn  ] New cabal version available. If you want to install this latest version, run 'ghcup install cabal 3.10.2.0'
[ Warn  ] New hls version available. If you want to install this latest version, run 'ghcup install hls 2.5.0.0'
[ Error ] [GHCup-07140] Both installation and setting the tool failed. Install error was: Unable to find a download for wasm32-wasi-9.9.20230623 
[ ...   ] Set error was: The version wasm32-wasi-9.9.20230623 of the tool ghc is not installed.
[ Error ] Also check the logs in /home/bradrn/.ghcup/logs
```

**Comment by hasufell** [comment]
> Unable to find a download for wasm32-wasi-9.9.20230623

The version doesn't exist: https://github.com/haskell/ghcup-metadata/blob/develop/ghcup-cross-0.0.8.yaml

**Comment by bradrn** [comment]
OK, now I’m really confused. The [instructions](https://www.haskell.org/ghcup/guide/#ghc-wasm-cross-bindists-experimental) suggest that I first build `wasm32-wasi-ghc` myself, then run that command to register it with GHCup. Accordingly I am using the `wasm32-wasi-ghc-9.9.20230623` that I have previously built. Yet that file suggests that I should instead be using a prebuilt bindist‽

…oh, actually, you know what, I just noticed: `export SKIP_GHC=yes`. So in fact this is _not_ asking me to build my own GHC: it’s using `ghc-wasm-meta` only to download the WASM toolchain, then using a prebuilt GHC. Is that correct?

**Comment by hasufell** [comment]
> OK, now I’m really confused. The instructions suggest that I first build wasm32-wasi-ghc myself, then run that command to register it with GHCup.

If you built a ghc via `ghcup compile ghc`, then you set it with `ghcup set ghc`. Compilation already installs. Running `ghcup install ghc` requires the versions to be in the metadata, unless you pass `ghcup install ghc -u <url>`.

I believe that is explained in `ghcup compile ghc --help` and `ghcup install ghc --help`.

**Comment by bradrn** [comment]
Clarification: when I said ‘build `wasm32-wasi-ghc` myself’, I meant *outside of GHCup*. Apologies for being unclear. Now that I think about it, there is no reason for GHCup to support a GHC binary built outside GHCup, so the behaviour I observed previously makes sense.

In any case, I followed the instructions more carefully, and it seems to be working fine now with `wasm32-wasi-9.8.0.20230927` — the bindist has installed without problems. Thanks for the help with this!
